pkg/segment/structs/segstructs.go

```diff
@@ -1196,14 +1197,20 @@ func (qa *QueryAggregators) IsStatsAggPresentInChain() bool {
 	return qa.HasInChain(statsAggPresentInCur)
 }
 
-func (nodeRes *NodeResult) StoreGlobalSearchError(errMsg string, logLevel log.Level) {
-	if nodeRes.GlobalSearchErrors == nil {
-		nodeRes.GlobalSearchErrors = make(map[string]*SearchErrorInfo)
+func (nodeRes *NodeResult) StoreGlobalSearchError(errMsg string, logLevel log.Level, err error) {
+	nodeRes.GlobalSearchErrors = StoreError(nodeRes.GlobalSearchErrors, errMsg, logLevel, err)
+}
```

Comment:
`errMsg` and `err` seem redundant. Can we just keep `err`, or is there a reason we need `errMsg` as well?

Reply:
`errMsg` is a common message, whereas `err` would have extra details about the error. We cannot always directly use `err` because it could be the case that somewhere we are printing record-specific details. If we map using this error, the keys could be unique and can result in a log flood. The reason for keeping err is that we can get extra details about this error if present and will be stored [once](https://github.com/siglens/siglens/blob/dd55833aba79270dbf78ad634d90dcd42683148e/pkg/segment/structs/segstructs.go#L1210).